File: Narek/week1/2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

char ShiftChar(char c);

int key=0;

int main(int argc, string argv[])
{
    int len;
    string plaintext;
    key = atoi(argv[1]);
    printf("Key: %i \n", key);
    plaintext = get_string("Enter text: ");
    len=strlen(plaintext);
    for(int i=0; i<len; i++)
    {
        char c=ShiftChar(plaintext[i]);
        printf("%c",c);
    }
    printf("\n");
}

char ShiftChar(char c){
    int a=0;
    if(isalpha(c)){
        if(islower(c)){
            a=97;
        }else{
            a=65;
        }
        int i = a + (c-a+key) % 26;
        return i;
    }
    else{
        return c;
    }
}

File: Harut/workspace/pset6/caesar/caesar.py
from cs50 import get_string
from sys import argv

def alpha(cc):
    if cc.islower():
        return ord(cc) - 97
    elif cc.isupper():
        return ord(cc) - 65


k = int(argv[1])
pt = get_string("enter plaintext: ")
ct = ''

for c in pt:
    if c.isupper():
        ct += chr((alpha(c) + k) % 26 + 65)

    elif c.islower():
        ct += chr((alpha(c) + k) % 26 + 97)

    else:
        ct += c

print(f'ciphertext: {ct}')